packages/flutter/lib/src/material/tabs.dart

```diff
@@ -922,12 +922,12 @@ class _TabBarViewState extends PageableListState<TabBarView> implements TabBarSe
       controller.value = scrollOffset / 2.0;
   }
 
-  Future fling(Offset scrollVelocity) {
+  Future fling(double scrollVelocity) {
     if (_selection == null || _selection.valueIsChanging)
       return new Future.value();
 
-    if (scrollVelocity.dx.abs() > _kMinFlingVelocity) {
-      final int selectionDelta = scrollVelocity.dx > 0 ? -1 : 1;
+    if (scrollVelocity.abs() > _kMinFlingVelocity) {
+      final int selectionDelta = scrollVelocity.sign.round();
```

Comment:
truncate() is probably more precisely the semantic we're going for here than round()